content/documents/configuring-user-application-data-access.mdx

```diff
@@ -0,0 +1,502 @@
+---
+type:           tutorial
+title:          Configuring User and Application Data Access
+description:    You have a newly created marketplace, what's next? Let's configure data access for your users and applications.
+author:         dsteinmetz
+publishDate:    2022-10-20
+updatedDate:    2022-10-20
+tags:           ["Security Profiles", "API Clients", "Tutorial"]
+---
+
+## Prerequisites
+In order to configure access to our marketplace data, we will need access to a marketplace. Create a Sandbox marketplace, or follow this tutorial to configure data access for an existing marketplace.
+
+## Assumptions
+- For this tutorial, we are using the URL for the sandbox environment in the region US-West.
+- You have an active buyer user with ID _and_ Username `buyeruser`, with a Password "Supersecurepassword123!"
+- You have an active seller user with ID _and_ Username `selleruser`, with a Password "Supersecurepassword123!"
+- Any ClientID value that is used in example requests should be replaced with the real ClientIDs that are created from making the requests in this tutorial. For our example, let's assume our ClientIDs are:
+    - Buyer storefront ClientID: `"00000000-0000-0000-0000-000000000000"`
+    - Admin backoffice ClientID: `"11111111-1111-1111-1111-111111111111"`
+    - Middlware API ClientID: `"222222222-2222-2222-2222-222222222222"`
+    - Middleware API Client Secret: `"supersecureclientsecretstring"`
+- You have created a new marketplace, or have access to an existing marketplace in the [OrderCloud Portal](https://portal.ordercloud.io)
+- You have and maintain a buyer storefront application, an admin backoffice application and a custom middleware API solution that handles more sensitive data and any custom integrations to OrderCloud
+
+## Creating API Clients
+Before we can configure data access for our applications, we need to create API Clients. API Clients are like more complex API keys that grants our different applications certain configured access to our marketplace data. From a security perspective, it is best practice to have one API Client per entry-point to your marketplace data.  
+
+### Create an API Client to represent a buyer storefront application
+First, lets create an API Client for our buyer storefront application.  This API Client will be the gateway for our buyer applicaiton to gain access to our marketplace data. We'll name it `Buyer Storefront`, designate it active, and make the access token duration 600 (minutes). Additionally, we will specify this API Client with `AllowAnyBuyer: true` which means that any buyer user will be able to successfully authenticate against this API Client. If a seller or supplier user attempted to authenticate against this API Client, they will get a `401 Unauthorized` response.
+<CodeExample
```

Comment:
> First, lets create an API Client

let's instead of lets

> gateway for our buyer applicaiton

application is spelled wrong

> If a seller or supplier user attempted

attempts instead of attempted